terraform-modules/aws/istio-networking/main-gateway/main.tf

```diff
@@ -24,6 +24,7 @@ resource "kubectl_manifest" "gateway" {
 
 # file templating
 data "template_file" "certificate" {
+  count     = ${var.enable_certificate 1 : 0}
   template = file("${path.module}/certificate.tpl.yaml")
```

Comment:
data component support count, this will avoid unnecessary data loading: [documentation](https://www.terraform.io/language/data-sources#:~:text=Data%20resources%20support%20count)